postreise/plot/plot_interconnection_map.py

```diff
@@ -1,249 +1,179 @@
         output_backend="webgl",
-        sizing_mode="stretch_both",
+        sizing_mode="scale_both",
         match_aspect=True,
     )
 
-    # for legend, hidden lines
-    leg_clr = ["#006ff9", "#8B36FF", "#01D4ED", "#FF2370"]
-    leg_lab = ["Western", "Eastern", "Texas", "HVDC"]
-    leg_xs = [-1.084288e07] * 4
-    leg_ys = [4.639031e06] * 4
-
-    for (colr, leg, x, y) in zip(leg_clr, leg_lab, leg_xs, leg_ys):
-        p.line(x, y, color=colr, width=5, legend=leg)
+    p.xgrid.visible = False
+    p.ygrid.visible = False
 
-    # pseudo lines for hover tips
-    lines = p.multi_line(
-        "xs", "ys", color="black", line_width="capacity", source=multi_line_source6
+    state_counts = count_nodes_per_state(grid)
+    plot_states(
+        bokeh_figure=p,
+        background_map=background_map,
+        state_list=list(state_counts["state"]),
+        labels_list=list(state_counts["count"]),
+        tooltip_name="nodes",
     )
 
-    # background tiles
-    p.add_tile(get_provider(Vendors.CARTODBPOSITRON))
-
-    # state borders
-    patch = p.patches("xs", "ys", fill_alpha=0.0, line_color="col", source=source)
-
-    # branches
-    source_list = [multi_line_source, multi_line_source2, multi_line_source3]
+    p.multi_line(
+        branch_west[["from_x", "to_x"]].values.tolist(),
+        branch_west[["from_y", "to_y"]].values.tolist(),
+        color="#006ff9",
+        line_width=branch_west["rateA"].abs() * branch_width_scale_factor,
+        legend_label="Western",
+    )
 
-    for (colr, source) in zip(leg_clr[0:3], source_list):
-        p.multi_line("xs", "ys", color=colr, line_width="capacity", source=source)
+    p.multi_line(
+        branch_east[["from_x", "to_x"]].values.tolist(),
+        branch_east[["from_y", "to_y"]].values.tolist(),
+        color="#8B36FF",
+        line_width=branch_east["rateA"].abs() * branch_width_scale_factor,
+        legend_label="Eastern",
+    )
 
     p.multi_line(
-        "xs", "ys", color="#FF2370", line_width="capacity", source=multi_line_source4
+        branch_tx[["from_x", "to_x"]].values.tolist(),
+        branch_tx[["from_y", "to_y"]].values.tolist(),
+        color="#01D4ED",
+        line_width=branch_tx["rateA"].abs() * branch_width_scale_factor,
+        legend_label="Texas",
     )
-    # pseudo ac
+
     p.multi_line(
-        "xs", "ys", color="col", line_width="capacity", source=multi_line_source5
+        dcline[["from_x", "to_x"]].values.tolist(),
+        dcline[["from_y", "to_y"]].values.tolist(),
+        color="#FF2370",
+        line_width=dcline["Pmax"].astype(float) * hvdc_width_scale_factor,
+        legend_label="HVDC",
     )
 
-    # triangles for b2b
     p.scatter(
-        x=b2b.from_x,
-        y=b2b.from_y,
+        x=b2b["from_x"],
+        y=b2b["from_y"],
         color="#FF2370",
         marker="triangle",
-        size=b2b.Pmax / 50 + 5,
-        legend="Back-to-Back",
+        size=b2b["Pmax"].astype(float) * b2b_size_scale_factor,
```

Comment:
Is there a problem with the default dtype?

Reply:
I don't think so. I guess I forgot to refactor this line

Reply:
Done